packages/page-template-modal/tsconfig.json

```diff
@@ -0,0 +1,32 @@
+{
+	"compilerOptions": {
+		"target": "es2017",
+		"module": "esnext",
+		"allowJs": true,
+		"checkJs": false,
+		"jsx": "react-jsx",
```

Comment:
I notice we use `"jsx": "react"` nearly everywhere else in calypso. I see it's a [simple difference](https://www.typescriptlang.org/docs/handbook/jsx.html), but I'm not sure about the implications - is this intentional?

Reply:
Yes this was intentional, even though you're right most places are using `jsx=react`.

In a file that's using jsx syntax, the compiler expects there to be an `import * as React from 'react'` at the top of the file. And that's what most of the code in Calypso etc. does. If you're going to write React code then you have to import React first.

However not all the ETK code was doing this. In particular, the layout modal wasn't importing React at the top of its modules. I wanted to leave this as it is because I figured it was something to do with the fact that as a WP plugin this wasn't using Facebook's `react`, but using `@wordpress/element` (or whatever our React wrapper is called). However leaving it the way it was caused a TypeScript error: you can't use jsx syntax in a file unless you've imported React first.

**Unless** tsconfig has `jsx=react-jsx` (or `react-jsxdev`). This option was introduced in React 17 when the requirement to import React first was dropped. Using this option makes the TypeScript compiler happy with the way that the layout modal code was already dealing with jsx syntax (whatever way that was)